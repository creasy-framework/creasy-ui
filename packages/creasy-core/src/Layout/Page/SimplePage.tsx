import React, { FunctionComponent, ReactElement } from 'react';
import classNames from 'classnames';
import { Scrollable } from '../../Interactive';
import { Card } from "../../Layout";
import { Spinner } from "../../Messaging";

interface Props {
  pageHeader?: ReactElement;
  pageFooter?: ReactElement;
  primaryContentHeader?: ReactElement;
  secondaryContentHeader?: ReactElement;
  contentFooter?: ReactElement;
  setScrollerRef?: (ref: any) => void;
  spinner?: ReactElement;
  isProcessing?: boolean;
  className?: string;
}

const SimplePage: FunctionComponent<Props> = ({
  pageHeader,
  pageFooter,
  primaryContentHeader,
  secondaryContentHeader,
  contentFooter,
  children,
  setScrollerRef,
  spinner = <Spinner size="m" type="donut" title="Loading..."/>,
  isProcessing,
  className,
}) =>  (
  <div className={classNames('creasy-page', className)}>
    {pageHeader && (
      <div className="creasy-page__page-header">
        {pageHeader}
      </div>
    )}
    <div className="creasy-page__scrollwrapper">
      <Scrollable className="creasy-page__scroller" setRef={setScrollerRef}>
        <div className="creasy-page__content">
          <div className="creasy-page__inner">
            {
              !isProcessing && (
                <Card
                  header={<Card.Header primary={primaryContentHeader} secondary={secondaryContentHeader}/>}
                  footer={contentFooter}
                >
                  {children}
                </Card>
              )
            }
            {isProcessing && (
              <div className="creasy-page__transition-spinner">
                {spinner}
              </div>
            )}
          </div>
        </div>
      </Scrollable>
    </div>
    {pageFooter && (
      <div className="creasy-page__page-footer">
        {pageFooter}
      </div>
    )}
  </div>
);

export default SimplePage;
