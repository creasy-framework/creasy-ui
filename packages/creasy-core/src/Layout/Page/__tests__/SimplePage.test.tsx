import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SimplePage from '../SimplePage';

describe('SimplePage', () => {
  const Header = () => <>Header</>;
  const Content = () => <>Content</>;
  const Footer = () => <>Footer</>;

  it('should render properly', () => {
    const wrapper = shallow(
      <SimplePage pageHeader={<Header/>} pageFooter={<Footer/>}>
        <Content/>
      </SimplePage>,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
