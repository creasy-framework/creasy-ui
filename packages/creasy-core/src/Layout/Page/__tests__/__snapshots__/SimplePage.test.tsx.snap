// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimplePage should render properly 1`] = `
<div
  className="creasy-page"
>
  <div
    className="creasy-page__page-header"
  >
    <Header />
  </div>
  <div
    className="creasy-page__scrollwrapper"
  >
    <Scrollable
      className="creasy-page__scroller"
    >
      <div
        className="creasy-page__content"
      >
        <div
          className="creasy-page__inner"
        >
          <Card
            header={<CardHeader />}
          >
            <Content />
          </Card>
        </div>
      </div>
    </Scrollable>
  </div>
  <div
    className="creasy-page__page-footer"
  >
    <Footer />
  </div>
</div>
`;
